packages/react-meteor-data/suspense/useFind.ts

```diff
@@ -0,0 +1,112 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { useEffect } from 'react'
+import {useFind as useFindClient} from '../useFind'
+import isEqual from 'lodash/isEqual'
+
+export const cacheMap = new Map<Mongo.Collection<unknown>, Entry[]>()
+interface Entry {
+  findArgs: Parameters<Mongo.Collection<unknown>['find']>
+  promise: Promise<unknown>
+  result?: unknown
+  error?: unknown
+  counter: number
+
+}
+export const removeFromArray =
+  <T>(list: T[], obj: T): void => {
+    if (obj) {
+      const index = list.indexOf(obj)
+      if (index !== -1) list.splice(index, 1)
+    }
+  }
+
+const removeNullCaches =
+  (cacheMap: Map<Mongo.Collection<unknown, unknown>, Entry[]>) => {
+    for (const cache of cacheMap.values()) {
+      cache
+        .filter(c => c.counter === 0)
+        .forEach(c => { removeFromArray(cache, c) })
+    }
+  }
+const useFindSuspense = <T = any>(
+  collection: Mongo.Collection<T>,
+  findArgs: Parameters<Mongo.Collection<T>['find']> | null
+) => {
+  useEffect(() => {
+    const cachedEntries = cacheMap.get(collection)
+    const entry = cachedEntries?.find(x => isEqual(x.findArgs, findArgs))
+    if (entry != null) ++entry.counter
+
+    removeNullCaches(cacheMap)
+    return () => {
+      // defer
+      setTimeout(() => {
+        const cachedEntries = cacheMap.get(collection)
+        const entry = cachedEntries?.find(x => isEqual(x.findArgs, findArgs))
+
+        if ((entry != null) && --entry.counter === 0) removeFromArray(cachedEntries, selector)
```

Comment:
We can make it simpler, as `removeNullCaches` will clean it anyway.
```suggestion
        if (entry != null) --entry.counter === 0;
```